fabfile.py

```diff
@@ -1,20 +1,161 @@
-from fabric.api import task, env, sudo, cd
+from fabric.api import task, env, sudo, cd, shell_env, require, put
+from functools import wraps
 
-tool_name = 'ifttt'
+import os
 
-env.hosts = ['tools-login.wmflabs.org']
-env.sudo_user = 'tools.{}'.format(tool_name)
-env.sudo_prefix = 'sudo -ni '
 env.use_ssh_config = True
+env.shell = '/bin/bash -c'
 
-home_dir = '/data/project/{}'.format(tool_name)
-code_dir = '{}/www/python/src'.format(home_dir)
+STAGES = {
+    'staging': {
+        'hosts': ['ifttt-staging.ifttt.eqiad.wmflabs'],
+        'local_config_file': './staging.cfg',
+        'branch': 'master',
+    },
+    'production': {
+        'hosts': ['ifttt-01.ifttt.eqiad.wmflabs'],
+        'local_config_file': './production.cfg',
+        'branch': 'master',
+    },
+}
+
+SOURCE_DIR = '/srv/ifttt'
+VENV_DIR = '/srv/ifttt/venv'
+DEST_CONFIG_FILE = 'ifttt.cfg'
+
+
+def sr(*cmd):
+    """
+    Sudo Run - Wraps a given command around sudo and runs it as the
+    www-data user
+    """
+    with shell_env(HOME='/srv/ifttt'):
+        return sudo(' '.join(cmd), user='www-data')
+
+
+def set_stage(stage='staging'):
+    """
+    Sets the stage and populate the environment with the necessary
+    config. Doing this allows accessing from anywhere stage related
+    details by simply doing something like env.source_dir etc
+
+    It also uses the values defined in the secrets directories
+    to substitute the templatized config files for the db and web configs,
+    and loads the final db, queue and web configs as strings into the
+    environment
+    """
+    env.stage = stage
+    for option, value in STAGES[env.stage].items():
+        setattr(env, option, value)
+
+
+def ensure_stage(fn):
+    """
+    Decorator to ensure the stage is set
+    """
+    @wraps(fn)
+    def wrapper(*args, **kwargs):
+        # The require operation will abort if the key stage
+        # is not set in the environment
+        require('stage', provided_by=(staging, production,))
+        return fn(*args, **kwargs)
+    return wrapper
+
+
+@task
+def production():
+    set_stage('production')
+
+
+@task
+def staging():
+    set_stage('staging')
+
+
+@task
+@ensure_stage
+def initialize_server():
```

Comment:
I think you should move the git clone to this as well, rather than leave that in puppet. We switched to that convention recently for ores.